js/classes/entities/midground/chad.js

```diff
@@ -1,58 +1,373 @@
-// SHELL FOR AN ENTITY. UPDATE AS WE GO.
+/**
+ * Chad is a mostly idle entity who must be able to walk for the tutorial.
+ * Otherwise he stands still in his idle position and offers dialog options to Chad.
+ * 
+ * @author Devin, Caleb, Nathan, Trae
+ */
 class Chad {
-    /**
-     * 
-     * @param {Vector} pos The y position of Chad (in the game world).
-     */
     constructor(pos) {
-        /** The position of Chad (in the game world). */
+        /** Checks if CHAD has collided with the ground. */
+        this.isOnGround = false;
+        /** Checks if CHAD can double jump. */
+        this.canDoubleJump = false;
+        /** Checks if CHAD has double jumped. */
+        this.hasDoubleJumped = false;
+        /** Checks if CHAD can dash. */
+        this.canDash = false;
+        /** Checks if CHAD has dashed. */
+        this.hasDashed = false;
+        /** Checks if CHAD is dashing. */
+        this.isDashing = false;
+        /** Gets the the y position of CHAD from the last time he was on the ground. */
+        this.prevYPosOnGround = 0;
+        /** The position of the Papa Chad (in the game world). */
         this.pos = pos;
-        /** An associative array of Chad's animations. Arranged [facing][action]. */
+        /** Gets the the x position of CHAD from the origin of where he started dashing. */
+        this.xDashAnchoredOrigin = 0;
+        /** An associative array of the animations for this Papa Chad. Arranged [facing][action]. */
         this.animations = [];
         this.loadAnimations();
-        /** What way is Chad looking? */
-        this.facing = "left"; // "left", "right"
-        /** What is Chad doing? */
+        /** What way is the Papa Chad looking? */
```

Comment:
There's still some Comments referring to Papa chad throughout the doc. I would use a ctrl-f to try and find all of them.